r2/r2/lib/base.py

```diff
@@ -43,12 +44,17 @@
 logging.getLogger('scgi-wsgi').setLevel(logging.CRITICAL)
 
 
-def is_local_address(ip):
-    if hasattr(g, 'trusted_proxy_ips'):
-        trusted_ips = [x.strip() for x in g.trusted_proxy_ips.split(',')]
-    else:
-        trusted_ips = []
-    return ip.startswith('10.') or ip == "127.0.0.1" or ip in trusted_ips
+def is_trusted_proxy(ip):
+    ip = ipaddress.IPv4Address(ip)
+    trusted_ips = getattr(g, 'trusted_proxy_ips', '').strip()
```

Comment:
This parsing in here is unnecessary if you use the config parsing layers, in `app_globals` you could add a section of `ConfigValue.tuple_of(ConfigValue.baseplate(ipaddress.IPv4Network))` (the baseplate thing is just to adapt the constructor of ipv4network to the expected protocol of those helpers) with `trusted_proxy_ips` (which should be renamed to `trusted_proxy_ranges` perhaps) and it'd handle parsing the comma-delimited line and converting to IPv4Network objects up-front rather than repeating it every request which is kinda messy and expensive (the ipaddress library is pretty slow).